src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +38,378 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
+    for (i = 0; i < ARRAY_COUNT(this->lightRaysColliders); i++) {
+        dim = &sQuadInit[i].dim;
+        Matrix_MultVec3f(&dim->quad[0], &v0);
+        Matrix_MultVec3f(&dim->quad[1], &v1);
+        Matrix_MultVec3f(&dim->quad[2], &v2);
+        Matrix_MultVec3f(&dim->quad[3], &v3);
+        collider = &this->lightRaysColliders[i];
+        Collider_SetQuadVertices(collider, &v0, &v1, &v2, &v3);
+    }
+
+    Matrix_Pop();
+}
+
+void BgIkanaMirror_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    BgIkanaMirror* this = THIS;
+    Vec3f* vtx;
+    Vec3f vertices[3];
+    s32 i;
+    s32 j;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gStoneTowerTempleMirrorCol);
+    Collider_InitTris(play, &this->mirrorCollider);
+    Collider_SetTris(play, &this->mirrorCollider, &this->dyna.actor, &sTrisInit, this->mirrorColliderElements);
+    Matrix_SetTranslateRotateYXZ(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
+                                 this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+
+    for (i = 0; i < ARRAY_COUNT(sTrisElementsInit); i++) {
+        vtx = &sTrisInit.elements[i].dim.vtx[0];
+        for (j = 0; j < 3; j++) {
+            Matrix_MultVec3f(&vtx[j], &vertices[j]);
+        }
+        Collider_SetTrisVertices(&this->mirrorCollider, i, &vertices[0], &vertices[1], &vertices[2]);
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->lightRaysColliders); i++) {
+        Collider_InitQuad(play, &this->lightRaysColliders[i]);
+        Collider_SetQuad(play, &this->lightRaysColliders[i], &this->dyna.actor, &sQuadInit[i]);
+    }
+
+    BgIkanaMirror_SetQuadVertices(this);
+    this->lightAbsorptionTexScroll = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightAbsorptionTexAnim);
+    this->lightEmissionTexScroll = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightEmissionTexAnim);
+    BgIkanaMirror_SetupWait(this);
+}
+
+void BgIkanaMirror_Destroy(Actor* thisx, PlayState* play) {
+    BgIkanaMirror* this = THIS;
+    s32 i;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(play, &this->mirrorCollider);
+
+    for (i = 0; i < ARRAY_COUNT(this->lightRaysColliders); i++) {
+        Collider_DestroyQuad(play, &this->lightRaysColliders[i]);
+    }
+}
+
+void BgIkanaMirror_SetupWait(BgIkanaMirror* this) {
+    this->isEmittingLight = 0;
+    this->actionFunc = BgIkanaMirror_Wait;
+}
+
+/*
+ * BgIkanaMirror_Wait is used to charge the mirror with light, increment the alpha value for the light absorption
+ * textures and decrement the alpha value for the light emission textures.
+ *
+ * By not directly setting the alpha values to 255 or 0, the transition from emitting light to
+ * absorbing it is visually smoother. Loading the proper display lists for absorption or emission of light is taken care
+ * of by BgIkanaMirror_Draw.
+ */
```

Comment:
```suggestion
/**
 * Charges the mirror with light, increment the alpha value for the light absorption textures and decrement the alpha value for the light emission textures.
 */
```
This is also rather verbose and redundant, this is enough. Also has to be a `/**` for Doxygen to pick it up (which is why we don't need the function name in here, if you hover over the function name in VSCode you'll see it attempt to produce a formatted Doxygen comment.